Index: hwinfo/src/hd/mouse.c
===================================================================
--- hwinfo.orig/src/hd/mouse.c
+++ hwinfo/src/hd/mouse.c
@@ -549,7 +549,8 @@ int _setspeed(int fd, int old, int new,
     default:    tty.c_cflag = flags | B1200; break;
     }
 
-  if(tcsetattr(fd, TCSAFLUSH, &tty)) return errno;
+  //remove serial port error
+  //if(tcsetattr(fd, TCSAFLUSH, &tty)) return errno;
 
   switch (new)
     {
@@ -566,7 +567,8 @@ int _setspeed(int fd, int old, int new,
 
   usleep(100000);
 
-  if(tcsetattr(fd, TCSAFLUSH, &tty)) return errno;
+  //remove serial port error
+  //if(tcsetattr(fd, TCSAFLUSH, &tty)) return errno;   
 
   return err;
 }
